### ISSUE TITLE
Edit/delete names of places/people for the contract journal

### ISSUE THREAD

**Issue by WiseWomanVP** [corona-warn-app/cwa-app-android#2079]
## Avoid duplicates
* [x] This enhancement request has not already been raised before
* [ ] Enhancement request is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)
* [ ] If you are proposing a new feature, please do so in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)

## Current Implementation

In the contract journal you can enter in names for people or places. However, if you happen to have the wrong keyboard currently active, after typing in a correct name ("Bio Company") it gets "corrected" to "Bin Company" by the keyboard between hitting "Done" and the name being stored. There is no way to correct or delete an entry, even one that is no longer needed for the past 14 days. 

## Suggested Enhancement

As in many other apps, I want to be able to press long on a name in the list and be able to correct the spelling. I also want to be able to delete names (as I now have 3 incorrect "Bin Company" entries as I was testing what happened).

## Expected Benefits

Helps me keep my list of people and places clean (I can delete or fix mistakes) and tidy.

**Comment by MikeMcC399** [comment]
@WiseWomanVP 
To edit, tap the three-dot overflow menu on the top right, select either
- Edit People or
- Edit Places

When the list of either People or Places is displayed, tap the pencil icon to edit.
When the edit screen is shown, there is also a bin icon to delete.

**Comment by WiseWomanVP** [comment]
Wow, that is non-intuitive ... If the three dots were at least on the page where I was entering them in. Okay, so the functionality is there, just well hidden. Thanks for the tip!

**Comment by MikeMcC399** [comment]
The German version of the Blog post on https://www.coronawarn.app/de/blog/2020-12-28-corona-warn-app-version-1-10/ has been updated to better explain about editing entries.

**Comment by MikeMcC399** [comment]
@WiseWomanVP 
Could you edit the title of your entry which has a typo "contract" instead of "contact" to help other people searching?

**Comment by WiseWomanVP** [comment]
Gah, sorry! Corrected.

**Comment by ndegendogo** [comment]
> Wow, that is non-intuitive

@WiseWomanVP yes, definitely. I also was missing this function, becsuse I started listing my contacts by first name only, but now I start to get duplicates.

**Comment by MikeMcC399** [comment]
@WiseWomanVP / @ndegendogo 
I've created a new wishlist item https://github.com/corona-warn-app/cwa-wishlist/issues/334 based on your valid feedback. Please feel free to add your comments there if I have missed anything in detail which you would like to have recommended.

**Comment by dsarkar** [comment]
@MikeMcC399 My question to you and the community is off-topic: In the PlayStore we had reports, that after upgrading to CWA 1.12, some users lost their contact journal entries. If you had entries pre-1.12, could you please check? Thanks.

**Comment by MikeMcC399** [comment]
@dsarkar 
> My question to you and the community is off-topic: In the PlayStore we had reports, that after upgrading to CWA 1.12, some users lost their contact journal entries. If you had entries pre-1.12, could you please check? Thanks.

This would really be better in a separate topic, but perhaps you thought you would reach an audience who was interesting in the Contact Journal?

The only point I found is that 1.12.0 keeps a history of 15 days, whereas 1.11.0 has one day less i.e. 14 days. So if somebody had entries for every day in 1.11.0 and then updated, it might look like the entries from the first day had been deleted. I made a suggestion in https://github.com/corona-warn-app/cwa-website/issues/893 about writing an FAQ article. I also found a text bug for which I will open a new issue.

**Comment by dsarkar** [comment]
@MikeMcC399 Thanks for the feedback.

> This would really be better in a separate topic

True.

**Comment by ndegendogo** [comment]
> The only point I found is that 1.12.0 keeps a history of 15 days, whereas 1.11.0 has one day less i.e. 14 days.

@MikeMcC399 @dsarkar same here on iOS. Was exactly 2 werks (14 entries) at 1.11, now one additional day (15 entries).

My guess was that this extension could be related to their decision to show the details of exposure warning on the diary pages. Corresponding to the checks (14 daily files plus the hourly files of today).

**Comment by MikeMcC399** [comment]
Was the original question answered? Can this issue be closed?

The wishlist item https://github.com/corona-warn-app/cwa-wishlist/issues/334 concerning editing / deleting is still open.

**Comment by WiseWomanVP** [comment]
Not really. There is a new update with additional data that I can store, but the problem of correcting or deleting an entry is still non-intuitive. Normally you would press long on an entra and shove it left to delete, or just a long press to correct. It is also problematic that the additional data always opens, instead of there being a little plus to open it.

**Comment by dsarkar** [comment]
@WiseWomanVP @MikeMcC399 @Ein-Tim  Is this suggestion Android specific from the UX point of view? Is there a substantial difference to the other existing proposals in the wishlist?

**Comment by MikeMcC399** [comment]
@dsarkar 
> Is this suggestion Android specific from the UX point of view?

I don't think it is Android-specific, so I will add the text from @WiseWomanVP onto https://github.com/corona-warn-app/cwa-wishlist/issues/334, so it can be considered in general.

**Comment by dsarkar** [comment]
Good morning @MikeMcC399, ok, I have seen it, and I copied your comment to your OP. We will track this issue here  with the same internal ticket.

Internal Tracking ID: [EXPOSUREAPP-3034](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-3034)
Internal Tracking ID: [EXPOSUREAPP-5495](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5495)
